### PR TITLE
e2e: basic multiple gc test

### PR DESCRIPTION
**Pull request by cnvergence**
**What type of PR is this?**

Adopted from https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/envoyproxy/gateway/issues/3364

### CODE REVIEW

**Comment by arkodg** [comment]
@cnvergence we could circumvent this issue by following what @shawnh2 did in https://github.com/envoyproxy/gateway/commit/decefccae531692b2e9b99929bf868deeb71b0e5 where he manually creates and applies the resources via the `client` API 😆

**Comment by cnvergence** [comment]
> @cnvergence we could circumvent this issue by following what @shawnh2 did in [decefcc](https://github.com/envoyproxy/gateway/commit/decefccae531692b2e9b99929bf868deeb71b0e5) where he manually creates and applies the resources via the `client` API 😆

I see 😄 and still ended up using suite client, I will then try this as well

File: test/e2e/multiple_gc/multiple_gc_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	e2e.CheckInstallScheme(t, c)
+
+	if flags.RunTest != nil && *flags.RunTest != "" {
+		t.Logf("Running E2E test %s with %s GatewayClass\n cleanup: %t\n debug: %t",
+			*flags.RunTest, *flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug)
+	} else {
+		t.Logf("Running E2E tests with %s GatewayClass\n cleanup: %t\n debug: %t",
+			*flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug)
+	}
+	t.Run("GC1Test", func(t *testing.T) {
```
Comment (arkodg) [inline test/e2e/multiple_gc/multiple_gc_test.go:48]: internet-gc-test so its clearer during debugging ?

File: test/e2e/multiple_gc/multiple_gc_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		// since this test suite reuse the base resources of previous test suite.
+		internetGatewaySuite.Applier.ManifestFS = []fs.FS{e2e.Manifests}
+		internetGatewaySuite.Applier.GatewayClass = internetGatewaySuiteGatewayClassName
+		internetGatewaySuite.ControllerName = kubernetes.GWCMustHaveAcceptedConditionTrue(t, internetGatewaySuite.Client, internetGatewaySuite.TimeoutConfig, internetGatewaySuite.GatewayClassName)
+
+		t.Logf("Running %d MultipleGC tests", len(tests.InternetGCTests))
+
+		err = internetGatewaySuite.Run(t, tests.InternetGCTests)
+		if err != nil {
+			t.Fatalf("Failed to run MergeGateways tests: %v", err)
```
Comment (arkodg) [inline test/e2e/multiple_gc/multiple_gc_test.go:76]:
```suggestion
			t.Fatalf("Failed to run InternetGC tests: %v", err)
```

File: test/e2e/multiple_gc/multiple_gc_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		t.Parallel()
+
+		privateGatewaySuiteGatewayClassName := "private"
+		if flags.RunTest != nil && *flags.RunTest != "" {
+			t.Logf("Running E2E test %s with %s GatewayClass\n cleanup: %t\n debug: %t",
+				*flags.RunTest, *flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug)
+		} else {
+			t.Logf("Running E2E tests with %s GatewayClass\n cleanup: %t\n debug: %t",
+				*flags.GatewayClassName, *flags.CleanupBaseResources, *flags.ShowDebug)
+		}
```
Comment (shawnh2) [inline test/e2e/multiple_gc/multiple_gc_test.go:90]: curious why we need this ? do we already have this check as in L41~L47

File: test/e2e/tests/tests.go [hunk 4]
```diff
@@ -14,4 +14,6 @@ var (
 	ConformanceTests   []suite.ConformanceTest
 	UpgradeTests       []suite.ConformanceTest
 	MergeGatewaysTests []suite.ConformanceTest
+	InternetGCTests    []suite.ConformanceTest
```
Comment (shawnh2) [inline test/e2e/tests/tests.go:17]: not a fan of this naming. should we just go as `MultipleGCTests` with type `map[string][]suite.ConformanceTest`, the GC name can be used as the map key ?

File: test/e2e/multiple_gc/multiple_gc_test.go (hunk 3 above)
Comment (cnvergence) [reply test/e2e/multiple_gc/multiple_gc_test.go:90]: good catch, I made mistake during refactoring this PR

File: test/e2e/tests/tests.go (hunk 4 above)
Comment (cnvergence) [reply test/e2e/tests/tests.go:17]: I agree, let's change the name for the tests, but I will keep this scenario naming just to keep up with the gw-api example

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**